packages/esm-patient-appointments-app/src/hooks/useAppointmentService.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, useMemo, useReducer } from 'react';
+import { getAppointmentServiceAll } from '../appointments/appointments.resource';
+import { Service } from '../types';
+
+enum ActionTypes {
+  pending = 'pending',
+  resolved = 'resolved',
+  error = 'error',
+}
+interface Pending {
+  type: ActionTypes.pending;
+}
+
+interface Error {
+  type: ActionTypes.error;
+  payload: Error;
+}
+
+interface Resolved {
+  type: ActionTypes.resolved;
+  payload: Array<Service>;
+}
+
+type Action = Pending | Error | Resolved;
+
+interface ServicePayload {
+  status: 'pending' | 'resolved' | 'error';
+  services: Array<Service>;
+  error?: null | Error;
+}
+
+function reducer(state: ServicePayload, action: Action): ServicePayload {
+  switch (action.type) {
+    case ActionTypes.pending:
+      return {
+        status: 'pending',
+        ...state,
+      };
+    case ActionTypes.resolved:
+      return {
+        status: 'resolved',
+        services: action.payload,
+        error: null,
+      };
+    case ActionTypes.error:
+      return {
+        status: 'error',
+        services: null,
+        error: action.payload,
+      };
+    default:
+      return state;
+  }
+}
+
+const useAppointmentService = () => {
+  const initialState: ServicePayload = { status: 'pending', services: [], error: null };
+  const [{ status, services, error }, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    const ac = new AbortController();
+    getAppointmentServiceAll(ac).then(
+      ({ data }) => {
+        dispatch({ type: ActionTypes.resolved, payload: data });
+      },
+      (error) => {
+        dispatch({ type: ActionTypes.error, payload: error });
+      },
+    );
+  }, []);
+
+  return useMemo(() => {
+    return { status, services, error };
+  }, [error, services, status]);
+};
+
+export default useAppointmentService;
```

Comment:
Not something that has to be changed now, but maybe something to consider depending on @denniskigen's migration plans:
I'd generally suggest to use SWR for every *new* data fetching hook that is added to the project to avoid later migrations and, most importantly, to reduce the number of different data fetching variants that we already have in the codebase. It would (probably) also make the hook a lot smaller. ;)

Reply:
I'll likely change this as part of the SWR migration but as this code doesn't exist yet, I'd prefer we don't do the change here. I'll base my work off of this code.

Reply:
Okay, great! cc @donaldkibet

Reply:
Oh wait, it looks like @donaldkibet already changed this.
@denniskigen With the code already being changed, would you prefer the changes being rolled back? If not, is there anything else that would have to be done in this area?

Reply:
Not particularly, this looks good enough to me!